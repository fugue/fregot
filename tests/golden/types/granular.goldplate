{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file"],
    "stdin": [
        ":l granular.rego",
        ":type people",
        ":type people[99].address",
        ":type [\"Hello\", 1, true]"
    ],
    "asserts": [
        {"stdout": "${GOLDPLATE_NAME}.stdout"},
        {"stderr": "${GOLDPLATE_NAME}.stderr"},
        {"exit_code": 0}
    ]
}
