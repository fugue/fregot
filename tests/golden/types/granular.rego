package granular

people = [
    {
        "name": "Tommy",
        "age": 32,
        "address": {
            "street": "Dubstrasse",
            "number": 66
        }
    }
]
