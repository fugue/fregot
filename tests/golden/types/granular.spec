{
    "command": "fregot",
    "arguments": ["repl", "--no-history-file"],
    "stdin": [
        ":l granular.rego",
        ":type people",
        ":type people[99].address"
    ],
    "asserts": [
        {"stdout": "${SPEC_NAME}.stdout"},
        {"stderr": "${SPEC_NAME}.stderr"},
        {"exit_code": 0}
    ]
}
