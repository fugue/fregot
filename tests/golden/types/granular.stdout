repl% granular% people [33m:[0m [34marray[0m[33m{[0m[34mobject[0m[33m{[0m
  [31m"name"[0m[33m:[0m [34mstring[0m[33m,[0m
  [31m"age"[0m[33m:[0m [34mnumber[0m[33m,[0m
  [31m"address"[0m[33m:[0m [34mobject[0m[33m{[0m
    [31m"number"[0m[33m:[0m [34mnumber[0m[33m,[0m
    [31m"street"[0m[33m:[0m [34mstring[0m
  [33m}[0m
[33m}[0m[33m}[0m
granular% people[33m[[0m[31m99[0m[33m][0m[33m.[0maddress [33m:[0m [34mobject[0m[33m{[0m
  [31m"number"[0m[33m:[0m [34mnumber[0m[33m,[0m
  [31m"street"[0m[33m:[0m [34mstring[0m
[33m}[0m
granular% 